lib/IRGen/IRGen.cpp

```diff
@@ -655,7 +656,7 @@ swift::createTargetMachine(IRGenOptions &Opts, ASTContext &Ctx) {
   std::string EffectiveClangTriple;
   std::vector<std::string> targetFeaturesArray;
   std::tie(TargetOpts, CPU, targetFeaturesArray, EffectiveClangTriple)
-    = getIRTargetOptions(Opts, Ctx);
+    = getIRTargetOptions(Ctx);
```

Comment:
Instead of using the tie formulation, if you just define a pair and then two references, you can use `const` here: 
```
const auto IRTargetOptions = getIRTargetOptions(Ctx);
const std::string &EffectiveClangTriple = IRTargetOptions.first;
const std::vector<std::string> &targetFeaturesArray  = IRTargetOptions.second;
```
Same number of lines and you can use `const`!